src/main/drivers/pwm_output.c

```diff
@@ -215,31 +203,31 @@ void pwmBrushlessMotorConfig(const timerHardware_t *timerHardware, uint8_t motor
     motors[motorIndex]->pwmWritePtr = pwmWriteStandard;
 }
 
-void pwmOneshotPwmRateMotorConfig(const timerHardware_t *timerHardware, uint8_t motorIndex, uint16_t motorPwmRate, uint16_t idlePulse, uint8_t useOneshot42)
+void pwmOneshotPwmRateMotorConfig(const timerHardware_t *timerHardware, uint8_t motorIndex, uint16_t motorPwmRate, uint8_t useOneshot42)
 {
-    uint32_t hz = ONESHOT_TIMER_MHZ * 1000000;
-    motors[motorIndex] = pwmOutConfig(timerHardware, ONESHOT_TIMER_MHZ, hz / motorPwmRate, idlePulse);
+    uint8_t mhz = ONESHOT125_TIMER_MHZ;
     if (useOneshot42) {
-        motors[motorIndex]->pwmWritePtr = pwmWriteOneshot42;
-    } else {
-        motors[motorIndex]->pwmWritePtr = pwmWriteOneshot125;
+        mhz = ONESHOT42_TIMER_MHZ;
```

Comment:
Looks good to me. If it is oneshot42 the mhz value already set for oneshot125 gets reinitialized with the related value for oneshot42. For sure you also can declare the variable and fill with an if else statement. Both ways work.